# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1318129109 25200
#      Sat Oct 08 19:58:29 2011 -0700
# Node ID db15303ee267b9e389941cad7ddfe63d72e08e6c
# Parent  b68b354a39c2ad4067b5c63ae9f86a868a8c34a9
substr.m: Fix failing %!test for 0 length.

* substr.m: Fix failing %!test for 0 length.

diff --git a/scripts/strings/substr.m b/scripts/strings/substr.m
--- a/scripts/strings/substr.m
+++ b/scripts/strings/substr.m
@@ -92,17 +92,17 @@ function t = substr (s, offset, len)
 endfunction
 
 
 %!assert (substr ("This is a test string", 6, 9), "is a test");
 %!assert (substr ("This is a test string", -11), "test string");
 %!assert (substr ("This is a test string", -11, 4), "test");
 %!assert (substr ("This is a test string", -11, -7), "test");
 %!assert (substr ("This is a test string", 1, -7), "This is a test");
-%!assert (substr ("This is a test string", 1, 0), "");
+%!assert (isempty (substr ("This is a test string", 1, 0)));
 
 %% Test input validation
 %!error substr ()
 %!error substr ("foo", 2, 3, 4)
 %!error substr (ones (5, 1), 1, 1)
 %!error substr ("foo", ones(2,2))
 %!error substr ("foo", 1, ones(2,2))
 %!error substr ("foo", 0)
